tests/common/plugins/custom_markers/__init__.py

```diff
@@ -46,11 +46,13 @@ def pytest_runtest_setup(item):
 def check_topology(item):
     # The closest marker is used here so that the module or class level
     # marker will be overrided by case level marker
-    toponames = item.get_closest_marker("topology")
-    if toponames:
+    topo_marks = [mark for mark in item.iter_markers(name="topology")]   # Get all 'topology' marks on the chain
+    if topo_marks:
+        topo_mark = topo_marks[0]   # The nearest mark overides others
         cfg_topos = item.config.getoption("--topology").split(',')
-        if all(topo not in toponames.args for topo in cfg_topos):
-            pytest.skip("test requires topology in {!r}".format(toponames))
+        if all(topo not in topo_mark.args for topo in cfg_topos):
+            pytest.skip("{} supports topologies: {}, specified topologies: {}"\
+                .format(item.nodeid, topo_mark.args, cfg_topos))
```

Comment:
My only concern with adding the nodeid is, it would make the log file more verbose. Now it would print the name of each test that is skipped (which is obviously useful) vs skipped[140]: <log message> for 140 testcases that had the same topology marker that didn't match the one that we passed in
@yxieca, what do you think?

Reply:
I agree with Neetha. I think we should trust explicitly skipped tests. We want to call out test cases that doesn't have any topology marker though.

Reply:
Sure, updated.